### ISSUE TITLE
Project does not have a license

### ISSUE THREAD

**Issue by TheWitheredStriker** [RMEx/scripts-externalizer#7]
I'm not sure if this is an oversight or deliberate, but the repository currently does not have a license. You may not be aware, but if a GitHub project does not have a license, [it means that forking and modifiying, or even USING the project _is actually illegal_.](https://choosealicense.com/no-permission/)

Since this repo has not been pushed to since 2018, I assume that you are no longer interested in developing this software. Since there is no license, it is currently completely impossible for someone else to take over, thereby dooming this project to be [abandonware](https://en.wikipedia.org/wiki/Abandonware).

This project is one of the greatest and most useful VX Ace scripts I have ever seen, and it would be a shame for it to end up in limbo like this. It would be incredible if others could update and refine the project, and perhaps backport it to XP and regular VX, but this will not legally be possible for as long as the repository does not have a license. For this reason, I politely ask of you to consider adding one. Some of the freer ones you could use are [MIT](https://choosealicense.com/licenses/mit/), [Boost](https://choosealicense.com/licenses/bsl-1.0/) and [BSD](https://opensource.org/license/bsd-3-clause/).

I hope you will read this and consider adding a license. This project is truly brilliant, and you have done an incredible job creating this. I hope you are well, wherever you may be now.

**Comment by xvw** [comment]
Since a lot of code and the concept come from RMEBuilder, it should be a GPL3 licence. We will discuss about it soon. Thanks

**Comment by TheWitheredStriker** [comment]
Alright, that's good to hear. Thank you for the quick response!

**Comment by xvw** [comment]
fixed on [e1558a9](https://github.com/RMEx/buildozer/commit/e1558a9d89d5bec886730d7c30f7a7b39c6b1f08)
Thanks !

**Comment by TheWitheredStriker** [comment]
Hello there, I wanted to bring to your attention that an adaptation of this script exists for RPG Maker XP, and can be found [here](https://github.com/TheNameIsAz/scripts-externalizer-XP). Just like this project before it, the script does not have a license. However, Joke/BilouMaster [has granted this adaptation their blessing/approval](https://save-point.org/thread-7612-post-51291.html#pid51291), long before this script was GPL'd.

The author of this adaptation has not been on GitHub since 2022, and I do not know where to find them. I assume that, since this repository is now licensed under the GPL, the XP adaptation must also be GPL'd? Or is it considered to already be under the GPL, now that the original codebase has been licensed as such? I am not a lawyer, so I do not know for certain. Again, the lack of a license on the XP adaptation makes the degree to which I can legally use and modify it rather blurry; thus why I ask.

**Comment by TheWitheredStriker** [comment]
@xvw It actually just hit me: since this script is now under the GPL, doesn't that mean that every RPG Maker game this is ever used in must also be licensed under the GPL?

If so, that would be very problematic, especially for commercial use.

Does Buildozer communicate at arm's length with the other Ruby code you would write for your game? If so, that would mean the GPL doesn't apply.

**Comment by xvw** [comment]
> @xvw It actually just hit me: since this script is now under the GPL, doesn't that mean that every RPG Maker game this is ever used in must also be licensed under the GPL?

Buildozer is a kind of Build system, so the "viral" aspect of the licence only applies if someone creates a build system based on Buildozer's source code (which is why Buildozer is licensed under GPL3, a large part of its code for "injecting Ruby" is based on [RMEBuilder](https://github.com/RMEx/RMEBuilder) code, which was also licensed under GPL3).

Even if Buildozer injects a tiny runtime, it is only used to orchestrate the build system, so it is not part of the game, so GPL3 does not apply to the game "produced via Buildozer". (Otherwise, all projects compiled with GCC should also be licensed under GPL3)

In a nutshell:
- Games that use Buildozer for development must **not** be licensed under GPL3 (because once the game is finished, Buildozer can be entirely removed from the project, but in any case, it's not necessary).
- If a project is inspired by Buildozer without taking code portions, it must **not** be licensed under GPL3.
- If a project is inspired by Buildozer by taking portions of the code, it **must** be licensed under GPL3.

So I guess there's no problem for commercial games?

**Comment by TheWitheredStriker** [comment]
Alright! That's great, actually. Thank you for the clarification!

**Comment by xvw** [comment]
https://github.com/RMEx/buildozer/issues/7#issuecomment-1651313634

Yes, it should also be licensed under GPL3. In an ideal world, Buildozer should be multi-version using the technique described by this script: [StandardizeRGSS](https://github.com/xvw/RPGMaker/blob/master/StandardizeRGSS/script.rb)

Unfortunately, I don't think anyone from the RMEx team is still active on RPGMaker.

**Comment by TheWitheredStriker** [comment]
It's fine. I plan to delve into RPG Maker (RGSS) development after I finish college and I would be happy to help out where possible (e.g. porting Buildozer to XP and regular VX; taking a crack at old RMEx projects; etc).

For now though, I don't have the time or skill for this. But I'm glad that this script still works fine, as it truly is a godsend imo. Once again, splendid job on (co-)creating this!

**Comment by xvw** [comment]
> It's fine. I plan to delve into RPG Maker (RGSS) development after I finish college and I would be happy to help out where possible (e.g. porting Buildozer to XP and regular VX; taking a crack at old RMEx projects; etc).

Thanks for the feedback and point out the absence of license!
(And happy to see an another Belgian interested in the project!)

_Veel succes met je studie. Tot snel_.

**Comment by TheWitheredStriker** [comment]
You're welcome!

_Bedankt, Xavier! Hopelijk tot binnenkort!_
